### PR TITLE
Use build.reason in template evaluation context

### PR DESCRIPTION
**Pull request by chcosta**
Checking to see if build.reason is available in template evaluation context yet..

### CODE REVIEW

File: .vsts-dotnet-ci.yml [hunk 1]
```diff
@@ -1,39 +1,43 @@
...
-        Build_Release:
-          _BuildConfig: Release
-          _PublishType: none
-          _SignType: test
+  - template: /eng/build.yml
+    parameters:
+      agentOs: Windows_NT
+      queue: 
+        ${{ if in(variables['build.Reason'], 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
+          name: Hosted VS2017
```
Comment (mmitche) [inline .vsts-dotnet-ci.yml]: This will change to "DotNetCore-Public" and "DotnetCore-Internal" when we have BYOC?

File: eng/build.yml [hunk 2]
```diff
@@ -19,37 +18,35 @@ phases:
...
     phaseName: ${{ parameters.agentOs }}
     dockerImage: ${{ parameters.dockerImage }}
     enableTelemetry: ${{ parameters.enableTelemetry }}
     fetchDepth: 5
     helixType: build/product
-    ${{ if notIn(parameters.buildReason, 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
-      helixSource: official/dotnet/arcade/$(Build.SourceBranch)
-    ${{ if in(parameters.buildReason, 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
+    ${{ if in(variables['build.Reason'], 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
       helixSource: pr/dotnet/arcade/$(Build.SourceBranch)
```
Comment (mmitche) [inline eng/build.yml]: @MattGal @chcosta Does this need to end in a /?

File: eng/common/templates/phases/base.yml [hunk 3]
```diff
@@ -34,10 +34,10 @@ phases:
       ${{ insert }}: ${{ parameters.phase.variables }}
   steps:
   - checkout: self
-    clean: true
+    clean: ${{ parameters.clean }}
```
Comment (mmitche) [inline eng/common/templates/phases/base.yml]: In what cases does this get overridden?

**Comment by mmitche** [comment]
> Doesn't look like variables are available yet in template evaluation context. Following up with VSTS folks to get an ETA..

When this happens, we'll use build-reason instead?

File: eng/common/templates/phases/base.yml (hunk 3 above)
Comment (chcosta) [reply eng/common/templates/phases/base.yml]: I'm not certain, I just realized there were places I was making assumptions about usage that weren't strictly required and it seems better to provide an overridable default.

File: eng/build.yml (hunk 2 above)
Comment (chcosta) [reply eng/build.yml]: @alexperovich ?

File: .vsts-dotnet-ci.yml (hunk 1 above)
Comment (chcosta) [reply .vsts-dotnet-ci.yml]: I'll take your word for it.  It's not clear to me how BYOC will plug into things and how exactly we'll interact with it.  The queue stuff (across repos), will need to change once BYOC is a thing though.

**Comment by chcosta** [comment]
I'd prefer to use build-reason, but https://dotnet.visualstudio.com/internal/_queries/edit/13/?triage=true, needs to be fixed first.  I closed that issue based on a discussion with Eric Sciple in which he said he fixed it and hotfixed it to our account, but I haven't actually validated that yet so I should probably re-open that issue until I do...

File: eng/build.yml (hunk 2 above)
Comment (alexperovich) [reply eng/build.yml]: Nothing about helix requires this to end in a '/' IIRC. The only requirement is that the view configuration match what this produces.

**Comment by chcosta** [comment]
After this merges and I validate official builds work, I'll merge `.vsts-dotnet-ci.yml` and `.vsts-dotnet.yml` into `.vsts-ci.yml` and update the corresponding build definitions.

**Pull request merged**